### PR TITLE
Adjustment Preflight Check for Redhat based systems and diff approach including RedHat specific vars

### PR DESCRIPTION
**Pull request by mto79**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

1. The preflight check in the roles/server/tasks/main.yml checks for RHEL 7 and RHEL 8 with ansible_distribution.
    According to https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html#ansible-facts-distribution it has to check against RedHat 7 and RedHat 8.
 
2. Also the use of spaces is not nice if you are using without quotes.
 
3. The checks with for e.g. ansible_distribution are out dated.
    image
    See also: https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html 

4. Including of vars only one time and using array variable for setting the checkmk requirements.
5. The loop has to be like below otherwise an array... must be list
```
- name: "Install Checkmk Prerequisites."
  ansible.builtin.package:
    name: "{{ item }}"
    state: present
  loop: "{{ checkmk_server_prerequisites }}" 
```
6. Added dnf-plugin-cores and yum-plugins-core in list variable  checkmk_server_prerequisites.. for later PR suggestion when looking at powertools/epel-release

## What is the new behavior?

- Check now on RedHat-7 and RedHat-8
- Use of '-' instead of space in the concat
- One inclusion of vars file
- Extension of checkmk_server_prequisities var

## Other information
https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html#ansible-facts-distribution
https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html

### CODE REVIEW

**Comment by mto79** [comment]
Well next try @robin-tribe29 and @diademiemi. Did an pull made repo as much as up-to-date. After that local linting and stupid me activated the Actions in my own repo .. so i am not bothering it here. Made the PR als smaller. No nifty nifty about ansible facts and the requested variable checkmk_server_prerequisites_per_distro instead of _checkmk_server_prerequisites. Today still testing against RHEL. But will keep that seperate, so i can keep PR as small. (Sorry haven to learn work flow on GitHub and be stricter with linting)

File: roles/server/tasks/main.yml [hunk 1]
```diff
@@ -65,15 +58,15 @@
...
+        msg: "{{ checkmk_server_verify_state.stdout_lines }}"
       when: checkmk_server_verify_setup | bool and ansible_os_family == "Debian"
 
 - name: "Import Checkmk GPG Key."
   become: true
   ansible.builtin.rpm_key:
     key: "/tmp/Check_MK-pubkey.gpg"
     state: present
-  when: checkmk_server_verify_setup | bool and ansible_os_family == "RedHat"
+  when: checkmk_server_verify_setup | bool and ansible_os_family. == "RedHat"
```
Comment (robin-tribe29) [inline roles/server/tasks/main.yml:69]: What does the ´.´ do here? Typo?

File: roles/server/tasks/main.yml [hunk 2]
```diff
@@ -4,26 +4,19 @@
...
   ansible.builtin.include_vars: "{{ ansible_os_family }}.yml"
   tags:
     - include-os-family-vars
 
-- name: "Include RHEL Version specific Variables."
-  ansible.builtin.include_vars: "{{ ansible_os_family }}-{{ ansible_distribution_major_version }}.yml"
-  when: ansible_os_family == "RedHat"
-  tags:
-    - include-rhel-version-vars
-
```
Comment (robin-tribe29) [inline roles/server/tasks/main.yml:19]: I still do not understand how the new variable structure is reflected within these tasks.
Can you explain, how this works?

File: roles/server/defaults/main.yml [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
-  - CentOS 7
-  - CentOS 8
-  - RHEL 7
-  - RHEL 8
+  - Debian-11
+  - Ubuntu-20
+  - CentOS-7
+  - CentOS-8
+  - RedHat-7
+  - RedHat-8
```
Comment (robin-tribe29) [inline roles/server/defaults/main.yml:9]: Still no fan of the looks of this, but I get why the proposal is a cleaner way.
@diademiemi what do you think regarding dropping `RHEL` in favor of `RedHat´?

File: roles/server/tasks/main.yml (hunk 1 above)
Comment (mto79) [reply roles/server/tasks/main.yml]: Fixed .

File: roles/server/tasks/main.yml (hunk 2 above)
Comment (mto79) [reply roles/server/tasks/main.yml:19]: The vars/RedHat.yml is (still) loaded and the it comes to:
```
checkmk_server_prerequisites_per_distro:
  RedHat:
    - epel-release
    - python3-libsemanage
    - dnf-plugins-core
  RedHat-7:
    - epel-release
    - libsemanage-python
    - yum-plugins-core
  RedHat-8:
    - epel-release
    - python3-libsemanage
    - dnf-plugins-core

checkmk_server_prerequisites: >-
  "{{ checkmk_server_prerequisites_per_distro[ansible_os_family+"-"+ansible_distribution_major_version] | default(
  checkmk_server_prerequisites_per_distro[ansible_os_family]) }}"
```
There the variable is being set by the RedHat version where running on. This saves you inclusion of apart vars files per repo. @diademiemi what do you say?

**Comment by robin-tribe29** [comment]
@mto79 I did not forget you, I promise. I am just still thinking about this and trying to plan for releases. I will keep you posted.

**Pull request merged**